src/server.rs

```diff
@@ -564,7 +564,10 @@ async fn handle_connection<C: Config>(
 
     let handshake: Handshake;
     handshake = c.dec.read_packet::<Handshake>().await?;
-    ensure!(handshake.server_address <= 255 || server.connection_mode() == ConnectionMode::Bungeecord, "server address too long");
+    ensure!(
+        handshake.server_address.len() <= 255 || server.connection_mode() == ConnectionMode::Bungeecord,
```

Comment:
`.len()` returns the byte length of the string, not a count of characters. `BoundedString` counts the characters.

`.chars().count()` should do the trick.